python/plugins/processing/tools/general.py

```diff
@@ -42,17 +42,36 @@
 
 
 def alglist(text=None):
+    print(get_alglist(text))
+
+    
+def algoptions(name):
+    s = get_algoptions(name)
+    if s:
+        print(s)
+    else:
+        print('Algorithm not found')
+    
+
+def alghelp(name):
+    s = get_alghelp(name)
+    if s:
+        print(s)
+    else:
+        print('Algorithm not found')
+    
+def get_alglist(text=None):
     s = ''
     for provider in list(algList.algs.values()):
         sortedlist = sorted(list(provider.values()), key=lambda alg: alg.name)
         for alg in sortedlist:
             if text is None or text.lower() in alg.name.lower():
                 s += alg.name.ljust(50, '-') + '--->' + alg.commandLineName() \
                     + '\n'
-    print(s)
+    return s
```

Comment:
Does this return a string now?
Wouldn't it be more appropriate to return a `list`?